### ISSUE TITLE
Incorrect cell selection if a frozen exists and the grid is scrolled down

### ISSUE THREAD

**Issue by mpegithub** [6pac/SlickGrid#329]
Hello,
In this [example example-frozen-columns-and-rows-spreadsheet.html](http://6pac.github.io/SlickGrid/examples/example-frozen-columns-and-rows-spreadsheet.html),
and in the bottom grid, if the grid is scrolled down and a selection is made, this one is incorrect: it looks like the number of scrolled lines are added to the selection...

**Comment by mpegithub** [comment]
I found a patch doing the job for me : in slick.cellrangeselector.js, function handleDrag

Replace :

```
     if ( (!_grid.canCellBeSelected( end.row, end.cell ) ) 
          || ( !_isRightCanvas && ( end.cell > _gridOptions.frozenColumn ) )
          || ( _isRightCanvas && ( end.cell <= _gridOptions.frozenColumn ) )
          || ( !_isBottomCanvas && ( end.row >= _gridOptions.frozenRow ) )
          || ( _isBottomCanvas && ( end.row < _gridOptions.frozenRow ) )
        ) {
         return;
        }
```

By :

```
        if (!_grid.canCellBeSelected(end.row, end.cell)) {
          return;
        }

        if (_gridOptions.frozenColumn >= 0) {
          if ((!_isRightCanvas && (end.cell > _gridOptions.frozenColumn)) ||
            (_isRightCanvas && (end.cell <= _gridOptions.frozenColumn))
          ) {
            return;
          }
        }

        if (_gridOptions.frozenRow >= 0) {
          if ((!_isBottomCanvas && (end.row >= _gridOptions.frozenRow)) ||
            (_isBottomCanvas && (end.row < _gridOptions.frozenRow))
          ) {
            console.log(end.row);
            return;
          }
        }
```

**Comment by ghiscoding** [comment]
I guess it could be simplified as 
```js
if (!_grid.canCellBeSelected(end.row, end.cell)) {
   return;
}

// when having frozen column(s), we need to do extra checks
if (_gridOptions.frozenColumn >= 0) {
  if (
   ((!_isRightCanvas && (end.cell > _gridOptions.frozenColumn)) || (_isRightCanvas && (end.cell <= _gridOptions.frozenColumn)))
   || ((!_isBottomCanvas && (end.row >= _gridOptions.frozenRow)) || (_isBottomCanvas && (end.row < _gridOptions.frozenRow)))
  ) {
   return;
 }
}
```

**Comment by ghiscoding** [comment]
ahh I see, you have separated the 2 types of frozen (columns/rows), I didn't realize that since both comments were very similar. Anyhow in both cases you want to `return` ASAP, so I don't think there's any performance difference in returning early (all in 1 big `if`). The way I wrote is 1 or the other, if both your `if` into 1 bigger `if`, though I didn't test anything

**Comment by mpegithub** [comment]
Yes I agree, maybe it can be useful to change the comment..

**Comment by ghiscoding** [comment]
yes this looks much better after a quick view at it. Have you tried your code with regular grid? I assume it's ok but we'll have to test that too. Can you make a PR for this and take the credits or you want us to do it?

**Comment by mpegithub** [comment]
What do you mean by regular grid ? Without frozen items ?
I am not used of github sorry, what is a PR ? No matter of who take the credits.

**Comment by ghiscoding** [comment]
Yes regular grid without any frozen options set. I think it's fine but we need to test a few options to make sure nothing breaks. 

If you could make the PR that would be great, but if don't know how or you can't do it then let us know and we'll take it

**Comment by mpegithub** [comment]
I understand, it seems to be ok for regular grids.
Sorry for the PR but i am a newbie here, thank you to do the job for me ;)

**Comment by ghiscoding** [comment]
ok might take a few days to look into this and testing it

**Comment by ghiscoding** [comment]
@mpegithub I tried your changes and that didn't work, however I went back to the source of what I copied it from, which is `X-Slickgrid` from [here](https://github.com/ddomingues/X-SlickGrid/blob/master/plugins/slick.cellrangeselector.js) and replaced some piece of code and now it works. Can you please test PR #331

**Comment by mpegithub** [comment]
Not working for me, sorry.

In my case I have no frozen row so :
1) _gridOptions.frozenRow = -1 => end.row is always >= _gridOptions.frozenRow 
2) active canvas is Not bottom canvas => !_isBottomCanvas is always true

=> row 114 is always true : ( !_isBottomCanvas && ( end.row >= _gridOptions.frozenRow ) )
=> always return

Or maybe I miss something !

**Comment by ghiscoding** [comment]
@mpegithub 
I pushed a fix, could you test the new fix and continue the conversation in the PR #331 
Thanks for all the feedback

**Comment by ghiscoding** [comment]
Closed by PR #331
